src/components/Button/Button.tsx

```diff
@@ -198,8 +207,29 @@ export function Button({
   const { Link } = useComponents();
   const LinkButton = BaseButton.withComponent(Link);
   const ButtonElement = props.href ? LinkButton : BaseButton;
+
+  /** Set default value for tracking dispatch */
+  const { label = undefined, component = 'button', data = undefined } =
+    props.trackingPayload || {};
+  const shouldTracking = label || props.enableTracking || false;
+
+  const dispatch = useClickTrigger();
+  const handler = (e: MouseEvent<any>): void => {
+    dispatch({
+      label,
+      component,
+      data
+    });
+    if (props.onClick) {
+      props.onClick(e);
+    }
+  };
+
   return (
-    <ButtonElement {...props}>
+    <ButtonElement
+      {...props}
+      onClick={shouldTracking ? handler : props.onClick}
```

Comment:
I feel this line destroy the "cleanliness" of this component. is there any other way to do this?

Reply:
You can delegate all to a single handler instead, but IMO it doesn't change that much